plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-order-downloads-controller.php

```diff
@@ -0,0 +1,267 @@
+			array(
+				'args'   => array(
+					'order_id' => array(
+						'description' => __( 'The order ID.', 'woocommerce' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check whether a given request has permission to read order downloads.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_post_permissions( $this->post_type, 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get downloadable items from an order.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function get_items( $request ) {
+		$order = wc_get_order( (int) $request['order_id'] );
+
+		if ( ! $order || $this->post_type !== $order->get_type() ) {
+			return new WP_Error( "woocommerce_rest_{$this->post_type}_invalid_id", __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		$downloads = $order->get_downloadable_items();
+
+		$data = array();
+		foreach ( $downloads as $download ) {
+			$download = $this->prepare_item_for_response( (object) $download, $request );
+			$download = $this->prepare_response_for_collection( $download );
+			$data[]   = $download;
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare a single download output for response.
+	 *
+	 * @param stdClass        $download Download object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $download, $request ) {
+		$data = array(
+			'download_id'         => $download->download_id,
+			'download_url'        => $download->download_url,
+			'product_id'          => $download->product_id,
+			'product_name'        => $download->product_name,
+			'download_name'       => $download->download_name,
+			'order_id'            => $download->order_id,
+			'order_key'           => $download->order_key,
+			'downloads_remaining' => '' === $download->downloads_remaining ? 'unlimited' : $download->downloads_remaining,
+			'access_expires'      => $download->access_expires ? wc_rest_prepare_date_response( $download->access_expires ) : 'never',
+			'access_expires_gmt'  => $download->access_expires ? wc_rest_prepare_date_response( get_gmt_from_date( $download->access_expires ) ) : 'never',
+			'file'                => $download->file,
+		);
```

Comment:
Are we intentionally not adding the `product_url` field? I realize consumers can derive it with a follow-up API call, but including it here seems easy and low overhead.